assets/src/edit-story/app/media/mediaProvider.js

```diff
@@ -73,11 +82,17 @@ function MediaProvider({ children }) {
         })
         .catch(fetchMediaError);
     },
-    [fetchMediaError, fetchMediaStart, getMedia, searchTerm]
+    [fetchMediaError, fetchMediaStart, getMedia]
   );
   const { uploadMedia, isUploading } = useUploadMedia({
-    media,
-    pagingNum,
+    mediaStateProvider: useCallback(
```

Comment:
And the same question here: it's not obvious that an uploading functionality should depend on the `mediaType`, `pageNum`, etc. The reverse could be true though.

Reply:
mediaType and pagingNum won't be needed here after https://github.com/google/web-stories-wp/pull/1848 has been submitted.

'media' doesn't really belong here, there should be a reducer action to update a single media resource, instead of ingesting and updating the entire media list.

This can be done as a separate change.